test/test_watch.rb

```diff
@@ -238,4 +238,82 @@ def test_watch_with_finish_and_ebadf
 
     assert_requested(:get, "#{api_host}/v1/watch/events", times: 1)
   end
+
+  def test_watch_finish_when_response_connection_open
```

Comment:
This is actually unrelated to the issue. I added it as that was my initial guess on where the issue is. Maybe it is still useful to have it here? But happy to revert this file changes.

Reply:
good to test weird edge-cases 👍